Readmessy (#67)
Roboy/ravestate: modules/ravestate_nlp/README.md

```

> (C) Roboy 2019            

Reviewer: ```suggestion
```
Maybe remove the (C)... These will get outdated really fast (even keeping the on ein the root README up to date will be a challenge)

| About Roboy                       | "nlp:roboy"  (property)       | Detecting whether sentence is about Roboy                                                 | 'you', 'roboy', 'robot', 'roboboy', ... |
| Yes-No                            | "nlp:yesno" (property)        | Detecting answers to yes-no questions                                                     | Checking for 'yes', 'no', 'i don't know', 'probably', 'probably not' and synonyms of these                           |
| Sentence Type: Question           | "nlp:is-question" (signal)    | Detecting whether sentence is a question                                                  |                                                |
| Play Game                         | "nlp:intent-play" (signal)    | Interlocutor wants to play a game                                                         | input: "I want to play", "I like games" or something similar    |

Additional features can be added and published to the system. All existing features can be found [here](__init__.py).

### Using the Features

#### React to Property Change
Each feature is stored in a ravestate property. 

> A state which wants to access a property needs to read permission fot that property.

Reviewer: ```suggestion
A state which wants to access a property needs read permissions for that property.
```

Example: State that reads the "yesno" property

```python
@state(
    cond=s("nlp:yesno:changed"),  # state reacts to a change in the 'yesno' property
    read="nlp:yesno",  # state is allowed to read the 'yesno' property

>     write="rawio:out")  # state is allowed to write to the output chanel 

Reviewer: ```suggestion
    write="rawio:out")  # state is allowed to write to the output property
```

def postive_chicken(ctx: ContextWrapper):
    if ctx["nlp:yesno"] == "yes":
        ctx["rawio:out"] = "You seem to be a positive chicken!"

```

#### React to Signal Emission
For 'Sentence Type: Question' a signal is emitted.

Example: State that reacts to the "is-question" signal 

```python
@state(
    cond=s("nlp:is-question"),  # state reacts to the is-question signal

>     write="rawio:out")  # state is allowed to write to the output chanel 

Reviewer: ```suggestion
    write="rawio:out")  # state is allowed to write to the output property 
```
